packages/network-controller/src/NetworkController.ts

```diff
@@ -443,25 +443,154 @@ export type NetworkControllerNetworkRemovedEvent = {
 };
 
 /**
- * `rpcEndpointUnavailable` is published after an attempt to make a request to
- * an RPC endpoint fails too many times in a row (because of a connection error
- * or an unusable response).
+ * `NetworkController:rpcEndpointChainUnavailable` is published when the number
+ * of failed consecutive attempts to receive a 2xx response from the primary
+ * endpoint of a chain of endpoints reaches a maximum, causing further requests
+ * to be temporarily paused, and when subsequent traffic to a failover endpoint
+ * similarly fails.
+ *
+ * In other words, this event will not published if a primary is deemed to be
+ * unavailable but its failover is not.
```

Comment:
Nit: I read this wrong the first time, here is an attempt at making it harder to misinterpret
```suggestion
 * In other words, this event will not published if the failover is available,
 * even if the primary is not.
```